kubernetes/cray-opa/templates/policies/keycloak-admin.yaml

```diff
@@ -79,6 +79,15 @@ data:
         required_admin_roles[r]
     }
 
+    # This checks the JWT if it has tenant-admin role
+    # also checks if the tenant header matches the group
+    allow {
+        roles_for_tenant[r]
```

Comment:
Looking roles_for_user, roles_for_admin, and roles_for tenant, they all look like duplicate code. Can these be refactored towards DRY to add clarity? 

Separately, while the tenant admin case is the only tenant persona we're tracking, please name symbols appropriately to ensure they track the tenant admin case. This should also help others with intuitive understanding of the policy.

Reply:
Based on how opa parses those separate sections are required. They are similar but not duplicate code, there is no way to combine those sections without major reworking existing multi-tenancy code.